trie/verkle.go

```diff
@@ -338,69 +342,55 @@ func addKey(s set, key []byte) {
-	if err != nil {
-		return fmt.Errorf("could not deserialize proof: %w", err)
-	}
-	cfg := verkle.GetConfig()
-	if !verkle.VerifyVerkleProof(proof, cis, indices, yis, cfg) {
-		return errInvalidProof
-	}
-
-	return nil
-}
+	rootC.SetBytesUnsafe(root)
 
-func deserializeVerkleProof(vp *verkle.VerkleProof, rootC *verkle.Point, statediff verkle.StateDiff) (*verkle.Proof, []*verkle.Point, []byte, []*verkle.Fr, error) {
 	var others set = set{} // Mark when an "other" stem has been seen
 
 	proof, err := verkle.DeserializeProof(vp, statediff)
 	if err != nil {
-		return nil, nil, nil, nil, fmt.Errorf("verkle proof deserialization error: %w", err)
+		return fmt.Errorf("verkle proof deserialization error: %w", err)
 	}
 
 	for _, stem := range proof.PoaStems {
 		addKey(others, stem)
 	}
 
-	if len(proof.Keys) != len(proof.Values) {
-		return nil, nil, nil, nil, fmt.Errorf("keys and values are of different length %d != %d", len(proof.Keys), len(proof.Values))
+	if len(proof.Keys) != len(proof.PreValues) {
+		return fmt.Errorf("keys and values are of different length %d != %d", len(proof.Keys), len(proof.PreValues))
 	}
 
-	tree, err := verkle.TreeFromProof(proof, rootC)
+	pretree, err := verkle.PreStateTreeFromProof(proof, rootC)
 	if err != nil {
-		return nil, nil, nil, nil, fmt.Errorf("error rebuilding the tree from proof: %w", err)
+		return fmt.Errorf("error rebuilding the pre-tree from proof: %w", err)
 	}
 	for _, stemdiff := range statediff {
 		for _, suffixdiff := range stemdiff.SuffixDiffs {
 			var key [32]byte
 			copy(key[:31], stemdiff.Stem[:])
 			key[31] = suffixdiff.Suffix
 
-			val, err := tree.Get(key[:], nil)
+			val, err := pretree.Get(key[:], nil)
 			if err != nil {
-				return nil, nil, nil, nil, fmt.Errorf("could not find key %x in tree rebuilt from proof: %w", key, err)
+				return fmt.Errorf("could not find key %x in tree rebuilt from proof: %w", key, err)
 			}
 			if len(val) > 0 {
 				if !bytes.Equal(val, suffixdiff.CurrentValue[:]) {
-					return nil, nil, nil, nil, fmt.Errorf("could not find correct value at %x in tree rebuilt from proof: %x != %x", key, val, *suffixdiff.CurrentValue)
+					return fmt.Errorf("could not find correct value at %x in tree rebuilt from proof: %x != %x", key, val, *suffixdiff.CurrentValue)
 				}
 			} else {
 				if suffixdiff.CurrentValue != nil && len(suffixdiff.CurrentValue) != 0 {
-					return nil, nil, nil, nil, fmt.Errorf("could not find correct value at %x in tree rebuilt from proof: %x != %x", key, val, *suffixdiff.CurrentValue)
+					return fmt.Errorf("could not find correct value at %x in tree rebuilt from proof: %x != %x", key, val, *suffixdiff.CurrentValue)
 				}
 			}
 		}
 	}
 
-	// no need to resolve as the tree has been reconstructed from the proof
-	// and must not contain any unresolved nodes.
-	pe, _, _, err := tree.GetProofItems(proof.Keys)
+	posttree, err := verkle.PostStateTreeFromProof(pretree, statediff)
```

Comment:
2nd part of making the proof: reconstruct the post tree from the pre-tree by copying the pre-tree and inserting all the `NewValues` from the proof into that post trie.